autonomx/AppModel.cpp

```diff
@@ -146,7 +154,10 @@ void AppModel::createGenerator() {
     }
 
     // create a new generator
-    QSharedPointer<Generator> generator = QSharedPointer<Generator>(new SpikingNet(nextID));
+    QSharedPointer<Generator> generator;
+    if (type.compare("SNN") == 0) {
```

Comment:
Perhaps we could compare to each element of a QList<QString> in a for loop.

Reply:
yes, i agree. i'm just not sure how we can construct the correct generator types without resorting to edge cases, as above... maybe this is what the `GeneratorFactory` was supposed to be used for? since the compared string argument is already supposed to be consistent across every part of the app where it's being used, we could absolutely locate the appropriate `GeneratorMeta` instance by key value and instantiate the class by pointing to its constructor dynamically. @jvlyndark what do you think?

Reply:
in any case, this is something we can resolve when we tie the Wolfram CA into this architecture, in a later PR.